### DISCUSSION TITLE
Start with .mkv file instead of optical media?

### DISCUSSION THREAD

**Discussion by SoulMinotaur515** [automatic-ripping-machine/automatic-ripping-machine#537] [General]
Hi folks, working on my first ARM setup and am slowly making progress.  I've a VMWare 6.7 instance in my home lab, a TrueNAS 12 file system, Plex for media, and now ARM on Ubuntu 20.04 (among a few other Ubuntu server VMs).
I read through the years worth of content on the https://b3n.org/automatic-ripping-machine/ page trying to find the answer but did not.  Okay, there was one entry from 2017, but that was 5 years ago.  
All of my current .mkv media I've converted manually using MakeMKV and whatever defaults it uses.  ARM does that as a first step, then moves it over to HandBrake to transcode prior to being put into the final folder.  I would like to be intentional about the final file and thus get those files to run through the HandBrake flow.  Is there a way to take one of my current .mkv files, drop them into the /raw folder and have ARM pick it up?  Alternately, what command would ARM use and I can just run it manually?
Thank you for any guidance.  I don't want to start re-encoding from optical if I can help it.

**Comment by 1337-server** [comment]
It's not currently possible to do this with ARM. It's something I do want to look into, but I've not had the time.

The command ARM uses is in the handbrake.py file

This is the cmd that gets sent to HandBrakeCLI
```
        cmd = f'nice {cfg.arm_config["HANDBRAKE_CLI"]} ' \
              f'-i {shlex.quote(srcpathname)} ' \
              f'-o {shlex.quote(filepathname)} ' \
              f'--preset "{hb_preset}" {hb_args} >> {logfile} 2>&1'
```

**Reply by SoulMinotaur515** [reply]
Thank you, I will manually transcode a few and see how that looks. If it seems worthwhile I will see about automating something. Powershell guy here so likely not something you could use directly in your python. 

Brainstorming here, since the idea has captured my attention.
Have a folder 'importmkv' or such, have ARM check it at an interval, and if a .mkv is found inject it into the existing flow at the same point as a file in 'raw' gets picked up. 
Rough concept, not sure where that fits with file names or media naming or BD vs DVD or other parts of ARM. 
BD-vs-DVD: check the media details and if it's resolution is < 1080 treat it like a DVD, and if > 1080 treat it like a BD.
File name: search for a title match on the existing file name? Require a text file having the same name as the .mkv, whose content is the correct name? I'm not sure what stage ARM looks for a title match. 

Again, dumping ideas from my brain, not necessarily solutions. 

I wish I had more time for hobbies, I'd love to try to make that work.

**Reply by 1337-server** [reply]
Batch converting should be simple enough with bash/powershell. More than likely it will be easier as there is no front end to worry about.

I do want to have ARM able to resume failed transcode, or convert a folder in the raw directory. But I haven't had a lot of free time at the moment, so work on ARM is rather slow. It should eventually be added, but I can't put a time on it.
